crosslink/README.md

```diff
@@ -0,0 +1,128 @@
+# Crosslink
+
+Crosslink is a tool to assist in managing go repositories that contain multiple
+intra-reposistory go.mod files. Crosslink automatically scans and inserts
+replace statements for direct and transitive intra-repository dependencies.
+Crosslink also contains functionality to remove any extra replace statements
+that are no longer required within reason (see below).
+
+## Rules
+
+Crosslink makes certain assumptions about your repository. Due to these
+assumptions the tool maintains contstraints to avoid any undesirable changes.
+
+1. By default all crosslink actions are non destructive unless specific command
+    flags are provided.
+2. Crosslink will only work with modules that fall under the root module
+    namespace.
+   - For example `example.com/crosslink/foo` falls under the
+    `example.com/crosslink` namespace but `example.com/bar` does not.
+   - Root module namespace is defined either by the module that exists in the
+    provided `--root` flag directory or the `go.mod` file located at highest
+    level of the repository.
+3. Crosslink does not maintain or include version numbers in replace
+   statements. Replace statements are always inserted or overwritten with no
+    version numbers.
+4. Crosslink allows users to `exclude` modules. Exclude means that crosslink
+   will not perform any replace or pruning operations where the
+   `old path == exclude path`. Operations will still be performed inside
+   modules where `module path == exclude path`.
+
+`Note:` Crosslink was developed for use in the OpenTelemetry organization.
+See [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go)
+and
+[opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib)
+for working examples. If you exerperience a use case that crosslink fails
+too handle properly please open an issue (or even a PR!) highlighting
+the discrepancy.
+
+## Usage
+
+### Latest Release
+
+To utilize the latest release of crosslink clone and build the repository and
+use the commands listed below.
+
+Crosslink is built as a Cobra app and supports the following commands and flags.
+
+### –-root
+
+Used to provide the path to a directory where a go.mod file must exist. If a
+root flag is not provided, crosslink will attempt to find the root directory of
+a git repository using a tool available in the go-build-tools repo. The root
+flag is available to all crosslink subcommands.
+
+**Note: If no --root flag is provided than crosslink attempts to identify a git
+repository in the current or a parent directory. If no git repository exists
+crosslink will panic.**
+
+    crosslink --root=/users/foo/multimodule-go-repo
+
+### prune / –-prune
+
+`CAUTION: DESTRUCTIVE`
+
+The prune command or flag will run the prune action on the current dependency.
```

Comment:
What does the "current dependency" mean here?

Reply:
I need to think on what I meant by this, it could be a leftover doc string from an earlier thought process. I will remove or update this.